lib/concurrent/at_exit.rb

```diff
@@ -0,0 +1,92 @@
+require 'concurrent/logging'
+require 'concurrent/synchronization'
+
+module Concurrent
+
+  # Provides ability to add and remove hooks to be run at `Kernel#at_exit`, order is undefined.
+  # Each hook is executed at most once.
+  class AtExitImplementation < Synchronization::Object
```

Comment:
It's not pedantic at all. I gladly add any improvement. It'll be here in a minute.